scripts/build.py

```diff
@@ -0,0 +1,243 @@
+  """
+  # vcpkg is a submodule of the main cpp repo
+  # Ensure that the submodule is initialized and updated
+  run_command(['git', 'submodule', 'init'])
+  run_command(['git', 'submodule', 'update'])
+
+  # Install vcpkg executable if its not installed already
+  vcpkg_executable_file_path = get_vcpkg_executable_file_path()
+  found_vcpkg_executable = os.path.exists(vcpkg_executable_file_path)
+  if not found_vcpkg_executable:
+    script_absolute_path = get_vcpkg_installation_script_path()
+    # mac example: ./external/vcpkg/bootstrap-sh
+    run_command([script_absolute_path])
+
+  # for each desktop platform, there is an existing vcpkg response file 
+  # in the repo (external/vcpkg_<triplet>_response_file.txt) defined for each target triplet
+  vcpkg_triplet = get_vcpkg_triplet(arch)
+  vcpkg_response_file_path = os.path.join(os.getcwd(), 'external', 
+                      'vcpkg_' + vcpkg_triplet + '_response_file.txt')
+
+  # mac example: ./external/vcpkg/vcpkg install @external/vcpkg_x64-osx_response_file.txt --disable-metrics
+  run_command([vcpkg_executable_file_path, 'install', '@' + vcpkg_response_file_path, '--disable-metrics'])
+
+
+def cmake_configure(build_dir, arch, build_tests=True, config=None):
+  """ CMake configure.
+
+  If you are seeing problems when running this multiple times,
+  make sure to clean/delete previous build directory.
+
+  Args:
+   build_dir (str): Output build directory.
+   arch (str): Platform Architecture (example: 'x64').
+   build_tests (bool): Build cpp unit tests.
+   config (str): Release/Debug config.
+          If its not specified, cmake's default is used (most likely Debug).
+  """   
+  cmd = ['cmake', '-S', '.', '-B', build_dir]
+  
+  # If generator is not specifed, default for platform is used by cmake, else 
+  # use the specified value
+  if config:
+    cmd.append('-DCMAKE_BUILD_TYPE={0}'.format(config))
+  if build_tests:
+    cmd.append('-DFIREBASE_CPP_BUILD_TESTS=ON')
+    cmd.append('-DFIREBASE_FORCE_FAKE_SECURE_STORAGE=ON')
+
+  vcpkg_toolchain_file_path = os.path.join(os.getcwd(), 'external', 'vcpkg', 'scripts', 'buildsystems', 'vcpkg.cmake')
+  cmd.append('-DCMAKE_TOOLCHAIN_FILE={0}'.format(vcpkg_toolchain_file_path))
+
+  vcpkg_triplet = get_vcpkg_triplet(arch)
+  cmd.append('-DVCPKG_TARGET_TRIPLET={0}'.format(vcpkg_triplet))
+  
+  # TODO: Remove this once firestore is included in the build and everything works
+  cmd.append('-DFIREBASE_INCLUDE_FIRESTORE=OFF')
+  run_command(cmd)
+
+
+def cmake_build(build_dir, targets=None):
+  """ CMake build.
+
+  Args:
+   build_dir (str): Output build directory.
+   cpu_cores (str): Number of cpu cores.
+   targets (list(str)): Specific build targets.
+             If not specified, cmake will build all targets.
+  """   
+  cmd = ['cmake', '--build', build_dir, '-j', str(os.cpu_count())]
+  if targets:
+    targets_cmd_line_arg = ['--target']
+    targets_cmd_line_arg.extend(args.target)
+    cmd.extend(targets)
+
+  run_command(cmd)
```

Comment:
There are a couple of issues with this method.

1. cpu_cores is no longer an arg.
2. 'args.target' is referenced, but args does not exist here.
3. 'targets_cmd_line_arg' is defined but not used. 

I'm guessing targets_cmd_line_arg is meant to be extended with 'targets' instead of args.target, and then cmd extended with targets_cmd_line_arg? Though you could skip the intermediate list and just do something like this:

cmd.append('--target')
cmd.extend(targets)

Reply:
Thanks for catching this. Yes, I changed the signature to remove `args` as an argument and missed this.